adapter/internal/eventhub/subscription.go

```diff
@@ -129,49 +158,34 @@ func LoadSubscriptionData(configFile *config.Config) {
 		queryParamMap := make(map[string]string, 1)
 		queryParamMap[GatewayLabelParam] = configuredEnv
 		go InvokeService(ApisEndpoint, APIListMap[configuredEnv], queryParamMap, APIListChannel, 0)
-	}
-
-	go retrieveAPIListFromChannel(APIListChannel)
-	var response response
-	for i := 1; i <= len(resources); i++ {
-		response = <-responseChannel
-
-		responseType := reflect.TypeOf(response.Type).Elem()
-		newResponse := reflect.New(responseType).Interface()
-
-		if response.Error == nil && response.Payload != nil {
-			err := json.Unmarshal(response.Payload, &newResponse)
-
-			if err != nil {
-				logger.LoggerSubscription.Errorf("Error occurred while unmarshalling the response received for: "+response.Endpoint, err)
+		for {
+			data := <-APIListChannel
+			logger.LoggerSync.Debug("Receiving data for an environment")
+			if data.Payload != nil {
```

Comment:
Should not we check if the data.error is nil as well.

Reply:
When there are no APIs deployed, there seems to be an error response saying no apis found for the criteria. I suppose this was the reason to avoid that condition for the fetch APIs call verification as well.